pkg/controller/controller.go

```diff
@@ -133,18 +136,46 @@ func (ctrl *CSIAttachController) Run(ctx context.Context, workers int) {
 		logger.Error(nil, "Cannot sync caches")
 		return
 	}
-	for i := 0; i < workers; i++ {
-		go wait.UntilWithContext(ctx, ctrl.syncVA, 0)
-		go wait.UntilWithContext(ctx, ctrl.syncPV, 0)
-	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.ReleaseLeaderElectionOnExit) {
+		for i := 0; i < workers; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				go wait.UntilWithContext(ctx, ctrl.syncVA, 0)
+			}()
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				go wait.UntilWithContext(ctx, ctrl.syncPV, 0)
+			}()
+		}
 
-	if ctrl.shouldReconcileVolumeAttachment {
-		go wait.UntilWithContext(ctx, func(ctx context.Context) {
-			err := ctrl.handler.ReconcileVA(ctx)
-			if err != nil {
-				logger.Error(err, "Failed to reconcile VolumeAttachment")
-			}
-		}, ctrl.reconcileSync)
+		if ctrl.shouldReconcileVolumeAttachment {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				go wait.UntilWithContext(ctx, func(ctx context.Context) {
```

Comment:
```suggestion
				wait.UntilWithContext(ctx, func(ctx context.Context) {
```

Otherwise the outer `go func()` finishes instantly.